library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/api/client/QuiltArmorMaterialExtensions.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.entity.api.client;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.item.ArmorMaterial;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+/**
+ * Extensions of {@link ArmorMaterial}, allowing for customizing the armor texture without having to use the
+ * {@link ArmorRenderingRegistry.TextureProvider} API.
+ */
+@InjectedInterface(ArmorMaterial.class)
+public interface QuiltArmorMaterialExtensions {
+	/**
+	 * Gets the base texture identifier to use for this armor material.
+	 * <p>
+	 * This will automatically have a {@code "_layer_1/2(_overlay).png"} suffix added to it, depending on the parameters
+	 * the vanilla game decides on.
+	 *
+	 * @see ArmorTextureUtils#getArmorTextureSuffix(boolean, String)
+	 * @return the base texture identifier
+	 */
+	// has a "default" impl in ArmorMaterialMixin
+	// not actually marked as default, since people NEED to implement this
+	@Environment(EnvType.CLIENT)
+	@NotNull Identifier getTexture();
```

Comment:
Since my last non-default injected interface methods (in Status Effect API) caused issues, should this be made default (and call `FallbackArmorTextureProvider.getArmorTexture` directly)?

Reply:
Well I went ahead and did that anyway, so